### ISSUE TITLE
Bug: Regression - setting bg color to NONE results in error

### ISSUE THREAD

**Issue by Theo-Steiner** [nvim-lualine/lualine.nvim#623]
### Self Checks

- [X] I'm using the latest lualine.
- [X] I didn't find the issue in exsisting issues or prs.

### How to reproduce the problem
paste this this inside init.vim
```lua
lua << END
local custom_auto = require'lualine.themes.auto'
-- Change the background of lualine_c section to NONE for normal mode
custom_auto.normal.c.bg = NONE

require'lualine'.setup  {
    options = {
        theme = custom_auto,
    }
}
END
```
### Expected behaviour
the background color in normal mode for section c will be none (transparent)

### Actual behaviour
error:
iE5108: Error executing lua ...m/plugged/lualine.nvim/lua/lualine/utils/color_utils.lua:286: attempt to perform arithmetic on a nil va
lue
stack traceback:
        ...m/plugged/lualine.nvim/lua/lualine/utils/color_utils.lua:286: in function 'get_color_distance'
        ...m/plugged/lualine.nvim/lua/lualine/utils/color_utils.lua:300: in function 'rgb2cterm'
        ...iner/.vim/plugged/lualine.nvim/lua/lualine/highlight.lua:125: in function 'highlight'
        ...iner/.vim/plugged/lualine.nvim/lua/lualine/highlight.lua:362: in function 'create_component_highlight_group'
        ...plugged/lualine.nvim/lua/lualine/components/filetype.lua:44: in function 'apply_icon'
        ...iner/.vim/plugged/lualine.nvim/lua/lualine/component.lua:218: in function 'draw'
        .../.vim/plugged/lualine.nvim/lua/lualine/utils/section.lua:26: in function 'draw_section'
        ...rs/theosteiner/.vim/plugged/lualine.nvim/lua/lualine.lua:152: in function 'statusline'
        ...rs/theosteiner/.vim/plugged/lualine.nvim/lua/lualine.lua:298: in function <...rs/theosteiner/.vim/plugged/lualine.nvim/lua/lualine.lua:283>

### Minimal config to reproduce the issue

<details>
<summary>Config</summary>

</details>

**Comment by shadmansaleh** [comment]
Try 
```lua
custom_auto.normal.c.bg = "None"
```

**Comment by Theo-Steiner** [comment]
I tried all of the following combinations just now:
```lua
custom_auto.normal.c.bg = "None"
```
+
```lua
custom_auto.normal.c.bg = 'None'
```
+
```lua
custom_auto.normal.c.bg = nil
```
All resulted in the same Error with the same stacktrace.
I also think I figured out what the error is: the ``rgb2cterm`` function is passed "#None" for "None", as ``sanitize_color`` in ``highlight.lua`` erronously assumes 'None' to be a hex color and therefore prefixes it with a # to '#None'.
By fixing ``sanitize_color`` using 'None', nil, NONE all work, although I don't quite understand what prevented nil from working in the first place.
